lib/user-methods.js

```diff
@@ -90,14 +90,17 @@ function fetchWithTimeout(input, init) {
                 url: xhr.responseURL,
                 headers: xhr.responseHeaders
             };
+            if (!options.headers) {
+                options.headers = {'content-type': xhr.getResponseHeader('content-type')};
+            }
             const body = 'response' in xhr ? xhr.response : xhr.responseText;
             resolve(new Response(body, options));
         };
         xhr.onerror = () => reject(new TypeError('Network request failed'));
         xhr.ontimeout = () => reject(new TypeError('Network request failed'));
```

Comment:
Similarly, it would be nice to reject that with a more meaningful message like `Network request timed out after xxx ms`.